### ISSUE TITLE
Test-Suite is broken

### ISSUE THREAD

**Issue by slyon** [httplib2/httplib2#166]
Dear httplib2-devs,

it seems like the test-suite of this package (in `python3/httplib2test.py`) is broken, due to the fact that it calls external APIs on servers which do not seem to exist anymore... (at `http://bitworking.org/projects/httplib2/test/`).

IMO it would be better to use a mock HTTP server, like https://github.com/gabrielfalcao/HTTPretty, which enables self-sustained testing.

For Debian/Ubuntu, I've worked on a patch to disable/skip most of the broken tests. But I've also started to fix some of those via the `HTTPretty` mock HTTP server. Please have a look at my patch and consider if using HTTPretty would be useful to improve the testability of your application:
https://git.launchpad.net/~slyon/ubuntu/+source/python-httplib2/tree/debian/patches/0003-Fix-autopkgtests.patch?h=fix-autopkgtest

**Comment by slyon** [comment]
Just saw the test-suite was reworked already!

**Comment by temoto** [comment]
Thanks a lot. I think we're fine with custom mock server over real sockets on localhost for now.

Files you are patching are no longer used, please see `tests/` directory.

**Comment by slyon** [comment]
Thanks for your quick reply!

Yeah... I just saw there are new `test/` and `tests/` directories, which are for some reason not part of the Debian package... Maybe the "old" tests (`python3/httplib2test.py`) should be removed, to avoid confusion between `test/`, `tests/` and `httplib2test.py`.

**Comment by temoto** [comment]
You are absolutely right, clean up is long overdue. At this point I will likely do this at once with `python2/` removal.
